### PR TITLE
Added basic JWT context with example.

### PR DESCRIPTION
**Pull request by rakshith-ravi**
No clue if it works or not.

@trezm I'm not even sure if I understand the way this is supposed to work. Do let me know if you want me to make any other changes.

### CODE REVIEW

**Comment by trezm** [comment]
Hey there @rakshith-ravi, this doesn't compile for me right now -- missing the `frank_jwt` dep and a few imports.

Beyond that, I'd recommend moving the custom context you've made into its own file, and then putting the whole thing in a `jwt_example` folder with a `main.rs`. That will help logically separate the code for newcomers looking at this example as well as keep the code a little cleaner :)

**Comment by rakshith-ravi** [comment]
Updated my PR. The example doesn't seem to compile for some reason. Fixed all other issues. However, it keeps expecting a `BasicContext` when I've clearly initialized it as a `JwtContext`. Could you take a look at it and see if I'm missing something?

File: thruster/examples/jwt_example/jwt_context.rs [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
+use std::str;
+use thruster::{
+    middleware::{
+        cookies::{Cookie, CookieOptions, HasCookies, SameSite},
+        query_params::HasQueryParams
+    },
+    Context, Request, Response,
+};
+
+pub struct JwtContext {
```
Comment (trezm) [inline thruster/examples/jwt_example/jwt_context.rs:12]: If you want to use a custom context, you must implement the `Context` trait, you can find that [here](https://github.com/trezm/Thruster/blob/master/thruster/src/core/context.rs).

File: thruster/examples/jwt_example/main.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    context.status(404);
+    context.body("Whoops! That route doesn't exist!");
+    Ok(context)
+}
+
+fn main() {
+    env_logger::init();
+    info!("Starting server...");
+
+    let mut app = App::<Request, JwtContext>::new_basic();
```
Comment (trezm) [inline thruster/examples/jwt_example/main.rs:114]: You'll also need to use a different method for creating the app, one that takes a context generator as an argument, so you'll need something like:

```rust
pub fn generate_context(request: Request) -> JwtContext {
    let mut ctx = JwtContext::new();
    ctx.params = request.params().clone();
    ctx.request = request;

    ctx
}

let mut app = App::<Request, JwtContext>:: create(generate_context);
```

File: thruster/examples/jwt_example/jwt_context.rs (hunk 1 above)
Comment (rakshith-ravi) [reply thruster/examples/jwt_example/jwt_context.rs:12]: Wait, but I am implementing `Context` though. Check out line number 134. I'm implementing `Context` for `JwtContext`

**Comment by rakshith-ravi** [comment]
Okay yeah. Made the changes. It compiles well now.

File: thruster/examples/jwt_example/jwt_context.rs (hunk 1 above)
Comment (trezm) [reply thruster/examples/jwt_example/jwt_context.rs:12]: My fault, for some reason the diff I was looking at didn't have passed line 21!

Comment (rakshith-ravi) [reply thruster/examples/jwt_example/jwt_context.rs:12]: No worries. Don't merge this yet. I'm noticing some issue on my side. I'll let you know once it's fixedx

**Comment by rakshith-ravi** [comment]
Okay, yeah. All fixed and working

**Comment by trezm** [comment]
Awesome, thanks so much for the contribution!! All merged in!

**Pull request merged**